src/replication.c

```diff
@@ -1792,10 +1808,27 @@ void readSyncBulkPayload(connection *conn) {
 
         /* RDB loading succeeded if we reach this point. */
         if (server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB) {
-            /* Delete the backup databases we created before starting to load
-             * the new RDB. Now the RDB was loaded with success so the old
-             * data is useless. */
-            disklessLoadDiscardBackup(diskless_load_backup, empty_db_flags);
+            /* Replica starts to apply data from new master, we must discard the cached
+            * master structure. */
+            serverAssert(server.master == NULL);
+            replicationDiscardCachedMaster();
+
+            /* We want our slaves to resync with us as well, if we have any sub-slaves.
+            * The master already transferred us an entirely different data set and we
+            * have no way to incrementally feed our slaves after that. */
+            disconnectSlaves(); /* Force our slaves to resync with us as well. */
+            freeReplicationBacklog(); /* Don't allow our chained slaves to PSYNC. */
```

Comment:
I think your change for #9398 is right, since, before `swap db`, all data is not changed. Maybe you could remove the same comments and add some new comments for explaining current `swapdb` mode special behaviors.

BTW, for redis multi line annotation, the format is
```
/*
 *
 */
instead of
/*
*
*/

'*' should be aligned 
```

Reply:
Done, extra change here is that this routine is now only executed for swapdb in case it's not asyncLoading.